WcaOnRails/app/models/competition.rb

```diff
@@ -473,6 +473,12 @@ def events_with_rounds_with_results
     end
   end
 
+  private def light_results_from_relation(relation)
+    ActiveRecord::Base.connection.execute(relation.to_sql).each(as: :hash).map do |r|
+      LightResult.new(r)
```

Comment:
I've "explained" it here: https://github.com/jfly/worldcubeassociation.org/pull/1/commits/743e0b1218d0bfed0bd2e988430348f8c6e7142a I didn't dive into ActiveRecord internals in order to figure out why exactly it's so slow, but creating and using thousands of ActiveRecord models has never been particularly fast.